src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -268,7 +268,17 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
             return $a == $b ? 0 : 1;
         });
 
-        if ($extra && $query = http_build_query($extra, '', '&')) {
+        // extract fragment
+        $fragment = isset($extra['_fragment']) ? $extra['_fragment'] : '';
+        unset($extra['_fragment']);
+
+        if (defined('PHP_QUERY_RFC3986')) {
+            $query = http_build_query($extra, '', '&', PHP_QUERY_RFC3986);
+        } else {
+            $query = http_build_query($extra, '', '&');
```

Comment:
Yeah 👍🏻. I pushed this in a rush earlier, I'll fix it shortly. 

Sent from my iPhone

> On 16 Aug 2016, at 20:49, Nicolas Grekas <notifications@github.com> wrote:
> 
> In src/Symfony/Component/Routing/Generator/UrlGenerator.php:
> 
> > @@ -268,7 +268,17 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
> >              return $a == $b ? 0 : 1;
> >          });
> >  
> > -        if ($extra && $query = http_build_query($extra, '', '&')) {
> > +        // extract fragment
> > +        $fragment = isset($extra['_fragment']) ? $extra['_fragment'] : '';
> > +        unset($extra['_fragment']);
> > +
> > +        if (defined('PHP_QUERY_RFC3986')) {
> > +            $query = http_build_query($extra, '', '&', PHP_QUERY_RFC3986);
> > +        } else {
> > +            $query = http_build_query($extra, '', '&');
> Can't we use str_replace here, from + to %20?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>